File: demo_python_tf/demo_python_tf/static_tf_broadcaster.py
import rclpy
from rclpy.node import Node
from tf2_ros import StaticTransformBroadcaster
from geometry_msgs.msg import TransformStamped
from tf_transformations import quaternion_from_euler
import math 

class StaticTFBroadcaster(Node):
    
    def __init__(self):
        super().__init__('static_tf_broadcaster')
        self.static_broadcaster = StaticTransformBroadcaster(self)
        self.publish_static_tf() 
    
    def publish_static_tf(self):
        
        """
        Broadcast coordinate relationship from base_link to camera_link
        """
        
        tranform = TransformStamped()
        tranform.header.frame_id = 'base_link'
        tranform.child_frame_id = 'camera_link'
        tranform.header.stamp = self.get_clock().now().to_msg()
        
        tranform.transform.translation.x = 0.5
        tranform.transform.translation.y = 0.3
        tranform.transform.translation.z = 0.6
        
        q = quaternion_from_euler(math.radians(180),0,0)

        tranform.transform.rotation.x = q[0]
        tranform.transform.rotation.y = q[1]
        tranform.transform.rotation.z = q[2]
        tranform.transform.rotation.w = q[3]
        
        self.static_broadcaster.sendTransform(tranform)
        self.get_logger().info(f"Broadcast tf: {tranform}")
        

def main():
    
    rclpy.init()
    node = StaticTFBroadcaster()
    rclpy.spin(node)
    rclpy.shutdown()
    

File: demo_cpp_tf/src/static_tf_broadcaster.cpp
#include "geometry_msgs/msg/transform_stamped.hpp"
#include "rclcpp/rclcpp.hpp"
#include "tf2/LinearMath/Quaternion.h"
#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
#include "tf2_ros/static_transform_broadcaster.h"


class StaticTFBroadcaster: public rclcpp::Node
{
private:
   std::shared_ptr<tf2_ros::StaticTransformBroadcaster> broadcaster_;

public:
    StaticTFBroadcaster():Node("static_tf_broadcaster"){
        broadcaster_ = std::make_shared<tf2_ros::StaticTransformBroadcaster>(this);
        this->publish_tf();
    }
    
    void publish_tf(){
        geometry_msgs::msg::TransformStamped transform;
        transform.header.stamp = this->get_clock()->now();
        transform.header.frame_id = "map";
        transform.child_frame_id = "target_point";
        transform.transform.translation.x = 5.0;
        transform.transform.translation.y = 3.0;
        transform.transform.translation.z = 0.0;
        tf2::Quaternion q;
        q.setRPY(0.0,0.0,60/180.0*M_PI);
        transform.transform.rotation = tf2::toMsg(q);
        this->broadcaster_->sendTransform(transform);
    }
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<StaticTFBroadcaster>();
    rclcpp::spin(node);
    return 0;
}

